pkg/icingadb/ha.go

```diff
@@ -155,10 +156,16 @@ func (h *HA) controller() {
 			}
 
 			shouldLog = false
-		case <-h.heartbeat.Lost():
+
+			if lost == nil {
+				lost = h.heartbeat.Lost()
+			}
```

Comment:
This is done too late here, still breaks if at least 60 seconds pass between the heartbeat event and reaching these lines. Moving this right at the top of this case would improve this (receiving from a channel and immediately calling a function probably won't take a minute).

All in all I'd prefer a solution where the heartbeat provided an interface where the obvious way to use it is correct and you don't have to build workarounds in the receiver.